runtime/lua/tohtml.lua

```diff
@@ -1291,7 +1291,29 @@ local function opt_to_global_state(opt, title)
   if opt.font then
     fonts = type(opt.font) == 'string' and { opt.font } or opt.font --[[@as (string[])]]
   elseif vim.o.guifont:match('^[^:]+') then
-    table.insert(fonts, vim.o.guifont:match('^[^:]+'))
+    --- @type string
+    local guifont = vim.o.guifont:match('^[^:]+')
+    local i = 1
+    local str, char, char_pre = '', '', ''
+    while i <= #guifont do
+      char = guifont:sub(i, i)
+      if char == ',' then
+        if char_pre == '\\' then
+          str = str .. char
+        else
+          table.insert(fonts, str)
+          str = ''
+        end
+      elseif char ~= '\\' and not (char == ' ' and char_pre == ',') then
```

Comment:
would `vim.iter` be helpful for simplifying this code?

need to find a way to simplify all of this. too hard to follow currently

Reply:
Hmm... I thought about it, but I couldn't come up with a good implementation. However, using for like this might make it a bit simpler:
```
local str, char_pre = '', ''
for _, char in ipairs(vim.split(vim.o.guifont:match('^[^:]+'), '')) do
    if char == ',' then
        if char_pre == '\\' then
            str = str .. char
        else
            table.insert(fonts, str)
            str = ''
        end
    elseif char ~= '\\' and not (char == ' ' and char_pre == ',') then
        str = str .. char
    end
    char_pre = char
end
table.insert(fonts, str)
```

Reply:
please also use `('...'):format()` where possible. string concat with `..` is harder to visualize, and often more verbose.

Reply:
please add a comment that shows a "input => output" example. that is 100x more readable than a bunch of string concatenations.